File: src/ssr_pkg/src/chao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <qq_msgs/Carry.h>

int main(int argc, char  *argv[])
{
    ros::init(argc, argv,"chao_node");
    printf("here is ros\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<qq_msgs::Carry>("kaihei",10);

    ros::Rate loop_rate(2);
    while (ros::ok())
    {
        qq_msgs::Carry msg;
        msg.grade = "青铜";
        msg.star = 999;
        msg.data = "国驸马朝2";
        pub.publish(msg);
        loop_rate.sleep();
    }
    
    return 0;
}

File: src/ssr_pkg/scripts/chao_node.py
#!/usr/bin/env python
#coding=utf8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry

if __name__ == '__main__':
    rospy.init_node('chao_node')
    rospy.loginfo('chao_node started')

    pub = rospy.Publisher("kaihei",Carry,queue_size=10)
    rate = rospy.Rate(2)

    while not rospy.is_shutdown():
        msg = Carry()
        msg.grade = "白银"
        msg.star = 100
        msg.data = "chao is here"
        pub.publish(msg)
        rate.sleep()
    